### PR TITLE
Closes #20 test data admiralneuro_adsl

### LINKED ISSUE #20: Test data: `admiralneuro_adsl`
Create a very simple `admiralneuro_adsl` based on `dm_neuro` with all vars needed for a standard BDS template to work (from [here](https://github.com/pharmaverse/admiral/blob/main/inst/templates/ad_advs.R), looks like TRTSDT. TRTESDT, TRT01A, TRT01P, etc).

### PR DESCRIPTION
**Pull request by Fanny-Gautier**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/CONTRIBUTING.html#detailed-development-process) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral family codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `main` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#deprecation)?
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` under the header `# admiral<ext> (development version)` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the Reference page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Comment by manciniedoardo** [comment]
@Fanny-Gautier do we actually want to provide need a template for `adsl` as well in `inst/templates`? If the intention is just to provide an `admiralneuro_adsl` dataset for use in ADPET templates, then I'd think having a script in `data-raw` that we don't expose to the users is enough?

@pharmaverse/admiralneuro thoughts?

**Comment by Fanny-Gautier** [comment]
> @Fanny-Gautier do we actually want to provide need a template for `adsl` as well in `inst/templates`? If the intention is just to provide an `admiralneuro_adsl` dataset for use in ADPET templates, then I'd think having a script in `data-raw` that we don't expose to the users is enough?
> 
> @pharmaverse/admiralneuro thoughts?

There's no need for the template. It was initially set up to create the ADaM in the usual way, but since it's only needed for ADPET, I can remove it.

File: data-raw/adsl_neuro.R [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+  definition = agegr1_lookup
+)
+
+# Derive Intent-to-Treat and Safety Population Flags ----
+adsl <- adsl %>%
+  mutate(ITTFL = "Y") %>%
+  derive_var_merged_exist_flag(
+    dataset_add = adsl,
+    by_vars = exprs(STUDYID, USUBJID),
+    new_var = SAFFL,
```
Comment (jwang-lilly) [inline data-raw/adsl_neuro.R:107]: AD0026: For subject-level character population flag variables: N = no (not included), Y = yes (included). Null values are not allowed.

File: data-raw/adsl_neuro.R [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+  TRTSTMF = "Time 1st Exposure Period Imput. Flag",
+  TRTEDTM = "Datetime of Last Exposure to Treatment",
+  TRTETMF = "Time Last Exposure Period Imput. Flag",
+  TRTSDT = "Date of First Exposure to Treatment",
+  TRTEDT = "Date of Last Exposure to Treatment",
+  TRTSTM = "Time of First Exposure to Treatment",
+  TRTDURD = "Treatment Duration",
+  AGEGR1 = "Age Group 1",
+  AGEGR1N = "Age Group 1 (N)",
+  ITTFL = "Intent-to-Treat Flag",
```
Comment (jwang-lilly) [inline data-raw/adsl_neuro.R:145]: ITTFL's label is Intent-to-Treat Population Flag

File: data-raw/adsl_neuro.R [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+  TRT01AN = "Actual Treatment for Period 01 (N)",
+  TRTSDTM = "Datetime of First Exposure to Treatment",
+  TRTSTMF = "Time 1st Exposure Period Imput. Flag",
+  TRTEDTM = "Datetime of Last Exposure to Treatment",
+  TRTETMF = "Time Last Exposure Period Imput. Flag",
+  TRTSDT = "Date of First Exposure to Treatment",
+  TRTEDT = "Date of Last Exposure to Treatment",
+  TRTSTM = "Time of First Exposure to Treatment",
+  TRTDURD = "Treatment Duration",
+  AGEGR1 = "Age Group 1",
```
Comment (jwang-lilly) [inline data-raw/adsl_neuro.R:143]: Pooled Age Group 1

File: data-raw/adsl_neuro.R [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+  AGEGR1N = "Age Group 1 (N)",
```
Comment (jwang-lilly) [inline data-raw/adsl_neuro.R:144]: Pooled Age Group 1 (N)

**Comment by jwang-lilly** [comment]
@Fanny-Gautier, great job with the adsl_neuro.  I completed my review, with a few minor comments.

File: data-raw/adsl_neuro.R (hunk 1 above)
Comment (Fanny-Gautier) [reply data-raw/adsl_neuro.R:107]: Good catch, `false_value` argument added

File: data-raw/adsl_neuro.R (hunk 2 above)
Comment (Fanny-Gautier) [reply data-raw/adsl_neuro.R:145]: updated as per IG v1.3

File: data-raw/adsl_neuro.R (hunk 3 above)
Comment (Fanny-Gautier) [reply data-raw/adsl_neuro.R:143]: thanks, updated as suggested and per IG v1.3

File: data-raw/adsl_neuro.R (hunk 4 above)
Comment (Fanny-Gautier) [reply data-raw/adsl_neuro.R:144]: thanks, updated as suggested and per IG v1.3

**Comment by jwang-lilly** [comment]
@Fanny-Gautier, perfect! Without further comments.

**Comment by manciniedoardo** [comment]
@jwang-lilly please can you approve? then I will merge

**Review by manciniedoardo — APPROVED** [review]
